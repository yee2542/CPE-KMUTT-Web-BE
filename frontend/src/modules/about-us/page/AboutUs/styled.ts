import styled from 'styled-components';

import BREAKPOINTS from 'common/constants/breakpoints';

export const AboutUsContainer = styled.div`
  display: flex;
  width: 100%;
  max-width: ${BREAKPOINTS.IPAD_LANDSCAPE};
  margin: 0 auto;
  justify-content: space-between;
  padding: 48px 0;
`;
