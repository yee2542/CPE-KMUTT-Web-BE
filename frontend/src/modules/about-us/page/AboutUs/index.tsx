import React, { useEffect, useState } from 'react';

import Head from 'next/head';

import CommonLayout from 'common/components/Layouts/CommonLayout';
import { STATIC_ABOUT_US_LINK, STATIC_HOME_LINK } from 'common/constants/links';

import ContentSection from 'modules/about-us/components/ContentSection';
import SidebarSection from 'modules/about-us/components/SidebarSection';
import { SIDEBAR_ITEMS } from 'modules/about-us/components/SidebarSection/constants';
import { HEADER_BACKGROUND_IMAGE } from 'modules/about-us/constants';
import { AboutUsContainer } from 'modules/about-us/page/AboutUs/styled';

const AboutUs: React.FC<{ id: string }> = ({ id }) => {
  const [aboutTitle, setAboutTitle] = useState(`เกี่ยวกับภาควิชาฯ`);

  const DEFAULT_NAVIGATE = [
    {
      title: 'หน้าแรก',
      link: STATIC_HOME_LINK,
    },
    {
      title: aboutTitle || 'เกี่ยวกับเรา',
      link: STATIC_ABOUT_US_LINK,
    },
  ];

  useEffect(() => {
    const title = SIDEBAR_ITEMS.map(
      ({ item }) => item.find(({ key }) => key === id)?.title,
    )
      .filter((element) => {
        return element !== undefined;
      })
      .toString();
    setAboutTitle(title);
  }, [id]);
  return (
    <>
      <Head>
        <title>ABOUT US | CPE KMUTT</title>
      </Head>

      <CommonLayout
        header="เกี่ยวกับเรา"
        navigate={DEFAULT_NAVIGATE}
        headerBackgroundImage={HEADER_BACKGROUND_IMAGE}>
        <AboutUsContainer>
          <SidebarSection activeKey={id} />
          <ContentSection />
        </AboutUsContainer>
      </CommonLayout>
    </>
  );
};

export default AboutUs;
