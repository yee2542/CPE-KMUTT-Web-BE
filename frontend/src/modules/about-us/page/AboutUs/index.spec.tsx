import React from 'react';

import { shallow } from 'enzyme';

import AboutUs from 'modules/about-us/page/AboutUs';
describe('AboutUs page', () => {
  it('should matched snapshots', () => {
    const wrapper = shallow(<AboutUs id="1" />);
    expect(wrapper).toMatchSnapshot();
  });
});
