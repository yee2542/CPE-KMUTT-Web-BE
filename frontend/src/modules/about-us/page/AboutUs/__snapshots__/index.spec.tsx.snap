// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs page should matched snapshots 1`] = `
<Fragment>
  <Head>
    <title>
      ABOUT US | CPE KMUTT
    </title>
  </Head>
  <CommonLayout
    header="เกี่ยวกับเรา"
    headerBackgroundImage="/images/about-us-banner.png"
    navigate={
      Array [
        Object {
          "link": "/",
          "title": "หน้าแรก",
        },
        Object {
          "link": "/about-us",
          "title": "เกี่ยวกับภาควิชาฯ",
        },
      ]
    }
  >
    <styled__AboutUsContainer>
      <SidebarSection
        activeKey="1"
      />
      <ContentSection />
    </styled__AboutUsContainer>
  </CommonLayout>
</Fragment>
`;
